### PR TITLE
Autotest cleanup generated files

### PR DESCRIPTION
**Pull request by khancyr**
Add LogMessages files to gitignore list
Put the generated files into buildlogs directory
fix logger_metadata generation on relative path

### CODE REVIEW

File: Tools/autotest/common.py [hunk 1]
```diff
@@ -1097,8 +1097,8 @@ def test_parameter_documentation_get_all_parameters(self):
             vehicle = "ArduCopter"
         if vehicle == "QuadPlane":
             vehicle = "ArduPlane"
-        cmd = [param_parse_filepath, '--vehicle', vehicle]
-#        cmd.append("--verbose")
+        cmd = [param_parse_filepath, '--vehicle', vehicle, "--format xml"]
```
Comment (peterbarker) [inline Tools/autotest/common.py:1100]: That should be
```suggestion
        cmd = [param_parse_filepath, '--vehicle', vehicle, "--format", "xml"]
```

... but I kind of like running the full generator here anyway.

File: .gitignore [hunk 2]
```diff
@@ -110,6 +110,9 @@ ParameterMetaData.xml
 apm.pdef.xml
 apm.pdef.json
 parameters.edn
+LogMessages.html
```
Comment (peterbarker) [inline .gitignore:113]: Could we instead move these into the buildlogs directory - or perhaps generate them there in the first place?

File: Tools/autotest/common.py (hunk 1 above)
Comment (khancyr) [reply Tools/autotest/common.py:1100]: ok done !

File: .gitignore (hunk 2 above)
Comment (khancyr) [reply .gitignore:113]: I think that is good to have on gitignore anyway but I have now move all generated files into buildlogs directory !

**Comment by khancyr** [comment]
I have move everything into the buildlogs directory. That need a little fix to logger_metadata

File: Tools/autotest/logger_metadata/parse.py [hunk 3]
```diff
@@ -82,6 +82,7 @@ def set_vehicles(self, vehicles):
     def search_for_files(self, dirs_to_search):
         _next = []
         for _dir in dirs_to_search:
+            _dir = os.path.join(topdir, _dir)
```
Comment (peterbarker) [inline Tools/autotest/logger_metadata/parse.py:85]: Have you tested parse.py when we're running the script which generates the logger documentation on the autotest server?

File: Tools/autotest/common.py [hunk 4]
```diff
@@ -1097,10 +1097,10 @@ def test_parameter_documentation_get_all_parameters(self):
             vehicle = "ArduCopter"
         if vehicle == "QuadPlane":
             vehicle = "ArduPlane"
-        cmd = [param_parse_filepath, '--vehicle', vehicle]
-#        cmd.append("--verbose")
+        cmd = [param_parse_filepath, '--vehicle', vehicle, '--format', 'all']
```
[the remark this answers is not in the document]
Comment (khancyr) [reply Tools/autotest/common.py:1100]: humm in fact it isn't need ... will revert

File: Tools/autotest/logger_metadata/parse.py (hunk 3 above)
Comment (khancyr) [reply Tools/autotest/logger_metadata/parse.py:85]: the autotest server should use `Tools/scripts/build_log_message_documentation.sh` that is call from the root of the Ardupilot git directory.
top_dir is correctly set as the root of ardupilot git directory, so it is good ! 
This modification allow to always get back to the root directory and thus find the vehicles subdirectory !

**Comment by peterbarker** [comment]
Do we still need the .gitignore changes?

**Comment by khancyr** [comment]
yes, if you manually run the loggermetadata parse you got them in you directory such as for the parametermetada

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**